apps/src/storage/dataBrowser/dataVisualizer/CrossTabChart.jsx

```diff
@@ -1,146 +1,188 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import {CROSS_TAB_CHART_AREA} from './constants';
+import * as color from '../../../util/color';
 
-const styles = {
-  headerCell: {
-    border: '1px solid black'
-  },
-  cell: {
-    height: '2em',
-    textAlign: 'center',
-    border: '1px solid black'
-  },
-  title: {
-    fontFamily: '"Gotham 5r", sans-serif, sans-serif',
-    fontSize: 16,
-    lineHeight: '16px',
-    color: 'black'
+export default function CrossTabChart(props) {
+  if (!props.records || !props.selectedColumn1 || !props.selectedColumn2) {
+    return null;
   }
-};
+  const {chartData, columns} = createPivotTable(
+    props.records,
+    props.numericColumns,
+    props.selectedColumn1,
+    props.selectedColumn2
+  );
+  const numericValues = [];
 
-class CrossTabChart extends React.Component {
-  static propTypes = {
-    records: PropTypes.array.isRequired,
-    numericColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
-    chartTitle: PropTypes.string,
-    selectedColumn1: PropTypes.string,
-    selectedColumn2: PropTypes.string
-  };
+  chartData.forEach(record => {
+    Object.entries(record).forEach(entry => {
+      let key = entry[0];
+      let value = entry[1];
+      if (typeof value === 'number' && key !== props.selectedColumn1) {
+        numericValues.push(value);
+      }
+    });
+  });
 
-  /**
-   * @param {Array.<Object>} records
-   * @param {string} rowName
-   * @param {string} columnName
-   */
-  createPivotTable = (records, rowName, columnName) => {
-    let countMap = {};
+  // Our goal is uniform column widths, for all columns except the first one.
+  // There are a few steps that lead to successful layout here.
+  // 1. The table width is 100% to maximize available space.
+  // 2. The first column uses `white-space: nowrap` so it won't shrink too
+  //    small for its contents.
+  // 3. We set the rest of the columns to have the same percentage-width,
+  //    adding up to 99% of the table width (which is more than available space)
```

Comment:
why 99 not 100?

Reply:
Only that it felt funny for these columns to add up to 100% when we're intentionally excluding the first column, and we can expect the first column to be at least 1% of the table width.  I think this would work perfectly well using 100% too, but 99% fit my intuition about what's going on better.

Reply:
Makes sense!